### PR TITLE
CDEC-481: Provide REST API for getting list of all users from on-prem

### PR DESCRIPTION
**Pull request by tolusha**

### CODE REVIEW

File: platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/utils/UserTransformer.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_PROFILE_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_REMOVE_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_UPDATE_PASSWORD;
+
+/**
+ * @author Anatoliy Bazko
+ */
+public class UserTransformer {
+
```
Comment (skabashnyuk) [inline platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/utils/UserTransformer.java]: Make this class similar to https://github.com/codenvy/che-core/blob/master/platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/DtoConverter.java

**Comment by tolusha** [comment]
@skabashnyuk @sleshchenko wdyt?

**Comment by evoevodin** [comment]
:+1:

**Comment by skabashnyuk** [comment]
ok

File: platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/LinksInjector.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+/**
+ * Helps to inject {@link UserService} related links.
+ *
+ * @author Anatoliy Bazko
+ */
+public final class LinksInjector {
+
+    public static UserDescriptor injectLinks(UserDescriptor userDescriptor, ServiceContext serviceContext) {
+        final User currentUser = EnvironmentContext.getCurrent().getUser();
+        final UriBuilder uriBuilder = serviceContext.getBaseUriBuilder();
```
Comment (sleshchenko) [inline platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/LinksInjector.java]: WDYT about using getServiceUriBuilder() method from ServiceContext and remove all .path(UserProfileService.class)?

**Comment by sleshchenko** [comment]
:+1:

**Pull request merged**